First pipeline run after scale-up fails

The first pipeline run after a scale-up from zero fails with `File not found: 'docker'`. Every further pipeline run works fine. 

```txt
Starting: Initialize job
Agent name: 'build-nme000001'
Agent machine name: 'build-nme000001'
Current agent version: '2.179.0'
Agent running as: 'AzDevOps'
Prepare build directory.
Set build variables.
Download all required tasks.
Downloading task: Bash (3.177.2)
Checking job knob settings.
Finished checking job knob settings.
##[error]File not found: 'docker'
Finishing: Initialize job

```

--- a/offerings/scalable/mainTemplate.json
+++ b/offerings/scalable/mainTemplate.json
@@ -48,14 +48,14 @@
             "metadata": {
                 "description": "The IP range of the vnet in CIDR notation."
             },
-            "defaultValue": "192.168.0.0/24"   
+            "defaultValue": "192.168.0.0/24"
         },
         "subnetPrefix": {
             "type": "string",
             "metadata": {
                 "description": "The IP range of the default subnet in CIDR notation."
             },
-            "defaultValue": "192.168.0.0/24"   
+            "defaultValue": "192.168.0.0/24"
         },
         "_artifactsLocation": {
             "type": "string",
@@ -74,7 +74,8 @@
     },
     "functions": [],
     "variables": {
-        "customData": "[concat('#include\n', uri(parameters('_artifactsLocation'), concat('scripts/cloud-init.yml', parameters('_artifactsLocationSasToken'))))]",
+        "scriptFolder": "scripts/",
+        "scriptFileName": "init-script.sh",
         "resourcesPrefix": "[uniqueString(resourceGroup().id)]",
         "vmssResourceName": "[concat(variables('resourcesPrefix'), '-', parameters('vmssName'))]",
         "singlePlacementGroup": "true",
@@ -164,8 +165,8 @@
                     "osProfile": {
                         "computerNamePrefix": "[parameters('vmssName')]",
                         "adminUsername": "[parameters('adminUsername')]",
-                        "adminPassword": "[parameters('adminPassword')]",
-                        "customData": "[base64(variables('customData'))]"                    },
+                        "adminPassword": "[parameters('adminPassword')]"
+                    },
                     "networkProfile": {
                         "networkInterfaceConfigurations": [
                             {
@@ -182,6 +183,25 @@
                                             }
                                         }
                                     ]
+                                }
+                            }
+                        ]
+                    },
+                    "extensionProfile": {
+                        "extensions": [
+                            {
+                                "name": "AppInstall",
+                                "properties": {
+                                    "publisher": "Microsoft.Azure.Extensions",
+                                    "type": "CustomScript",
+                                    "typeHandlerVersion": "2.0",
+                                    "autoUpgradeMinorVersion": true,
+                                    "settings": {
+                                        "fileUris": [
+                                            "[uri(parameters('_artifactsLocation'), concat(variables('scriptFolder'), variables('scriptFileName'), parameters('_artifactsLocationSasToken')))]"
+                                        ],
+                                        "commandToExecute": "[concat('sh ', variables('scriptFileName'))]"
+                                    }
                                 }
                             }
                         ]

--- a/offerings/scalable/scripts/init-script.sh
+++ b/offerings/scalable/scripts/init-script.sh
@@ -1,0 +1,14 @@
+#!/bin/sh
+
+apt-get install apt-transport-https -y \
+ca-certificates -y \
+curl -y \
+gnupg-agent -y \
+software-properties-common -y
+
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+apt-get update -y 
+apt-get install -y docker-ce=5:20.10.0~3-0~ubuntu-bionic docker-ce-cli=5:20.10.0~3-0~ubuntu-bionic containerd.io=1.4.3-1
+systemctl start docker
+systemctl enable docker

